cardano-crypto-class/src/Cardano/Crypto/PinnedSizedBytes.hs

```diff
@@ -0,0 +1,229 @@
+{-# LANGUAGE BangPatterns        #-}
+{-# LANGUAGE DerivingVia #-}
+{-# LANGUAGE DataKinds           #-}
+{-# LANGUAGE KindSignatures      #-}
+{-# LANGUAGE MagicHash           #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+{-# LANGUAGE UnboxedTuples       #-}
+
+-- for pinnedByteArrayFromListN
+{-# OPTIONS_GHC -Wno-missing-local-signatures #-}
+module Cardano.Crypto.PinnedSizedBytes
+  (
+    PinnedSizedBytes,
+    -- * Initialization
+    psbZero,
+    -- * Conversions
+    psbFromBytes,
+    psbToBytes,
+    psbFromByteString,
+    psbFromByteStringCheck,
+    psbToByteString,
+    -- * C usage
+    psbUseAsCPtr,
+    psbUseAsSizedPtr,
+    psbCreate,
+    psbCreateSized,
+    ptrPsbToSizedPtr,
+  ) where
+
+import Control.Monad.ST (runST)
+import Control.Monad.Primitive  (primitive_)
+import Data.Char (ord)
+import Data.Primitive.ByteArray (ByteArray (..), MutableByteArray (..), copyByteArrayToAddr, newPinnedByteArray, unsafeFreezeByteArray, foldrByteArray, byteArrayContents, writeByteArray, mutableByteArrayContents)
```

Comment:
[style] long lines!

Reply:
I asked about code style when working on the original PR, I got reply "there isn't".

Reply:
Fixed.